src/main/java/com/orbitz/consul/config/CacheConfig.java

```diff
@@ -0,0 +1,133 @@
+package com.orbitz.consul.config;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+
+public class CacheConfig {
+
+    @VisibleForTesting
+    static final Duration DEFAULT_WATCH_DURATION = Duration.ofSeconds(10);
+    @VisibleForTesting
+    static final Duration DEFAULT_BACKOFF_DELAY = Duration.ofSeconds(10);
+    @VisibleForTesting
+    static final Duration DEFAULT_MIN_DELAY_BETWEEN_REQUESTS = Duration.ZERO;
+    @VisibleForTesting
+    static final boolean DEFAULT_TIMEOUT_AUTO_ADJUSTMENT_ENABLED = true;
+    @VisibleForTesting
+    static final Duration DEFAULT_TIMEOUT_AUTO_ADJUSTMENT_MARGIN = Duration.ofSeconds(2);
+
+    private final Duration backOffDelay;
+    private final Duration minDelayBetweenRequests;
+    private final Duration timeoutAutoAdjustmentMargin;
+    private final boolean timeoutAutoAdjustmentEnabled;
+
+    private CacheConfig(Duration backOffDelay, Duration minDelayBetweenRequests,
+                        boolean timeoutAutoAdjustmentEnabled, Duration timeoutAutoAdjustmentMargin) {
+        this.backOffDelay = backOffDelay;
+        this.minDelayBetweenRequests = minDelayBetweenRequests;
+        this.timeoutAutoAdjustmentEnabled = timeoutAutoAdjustmentEnabled;
+        this.timeoutAutoAdjustmentMargin = timeoutAutoAdjustmentMargin;
+    }
+
+    /**
+     * Gets the default watch duration for caches.
+     */
+    public Duration getWatchDuration() {
+        return DEFAULT_WATCH_DURATION;
+    }
+
+    /**
+     * Gets the back-off delay used in caches.
+     */
+    public Duration getBackOffDelay() {
+        return backOffDelay;
+    }
+
+    /**
+     * Is the automatic adjustment of read timeout enabled?
+     */
+    public boolean isTimeoutAutoAdjustmentEnabled() {
+       return timeoutAutoAdjustmentEnabled;
+    }
+
+    /**
+     * Gets the margin of the read timeout for caches.
+     * The margin represents the additional amount of time given to the read timeout, in addition to the wait duration.
+     */
+    public Duration getTimeoutAutoAdjustmentMargin() {
+        return timeoutAutoAdjustmentMargin;
+    }
+
+    /**
+     * Gets the minimum time between two requests for caches.
+     */
+    public Duration getMinimumDurationBetweenRequests() {
+        return minDelayBetweenRequests;
+    }
+
+    /**
+     * Creates a new {@link CacheConfig.Builder} object.
+     *
+     * @return A new Consul builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private Duration backOffDelay;
+        private Duration minDelayBetweenRequests;
+        private Duration timeoutAutoAdjustmentMargin;
+        private Boolean timeoutAutoAdjustmentEnabled;
+
+        public Builder() {
+
+        }
```

Comment:
This comes as a duplicate of `CacheConfig.builder()` call. It could be private to have one and only one way to create a builder.

Reply:
done.